File: algae-cppserver/src/main/cpp/sourceLocation.cpp
/**
 * sourceLocation.cpp
 *
 *
 *  Created on: July 3, 2012
 *      Author: zeil
 */

#include <algae/snapshot/sourceLocation.h>

#include <string>

namespace algae {

SourceLocation::SourceLocation (std::string filename, int line)
: fileName(filename), lineNumber(line)
{
}

SourceLocation::SourceLocation ()
: fileName(""), lineNumber(1)
{
	fileName = "";
	setLineNumber(1);
}


void SourceLocation::print(std::ostream& out) const {
	out << fileName << ":" << lineNumber;
}

bool SourceLocation::operator== (const SourceLocation& other) const {
	if (getLineNumber() != other.getLineNumber())
		return false;
	else return fileName == other.fileName;
}

bool SourceLocation::operator< (const SourceLocation& other) const {
	if (fileName < other.fileName)
		return true;
	if (fileName > other.fileName)
		return false;
	return (getLineNumber() < other.getLineNumber());
}






}

File: algae-client-server/src/main/java/edu/odu/cs/AlgAE/Common/Snapshot/SourceLocation.java
package edu.odu.cs.AlgAE.Common.Snapshot;

public class SourceLocation {

    private String fileName;
    private int lineNumber;
    
    public SourceLocation (String filename, int line) {
        fileName = filename;
        setLineNumber(line);
    }
    
    public SourceLocation () {
        fileName = "";
        setLineNumber(1);
    }
    
    
    public String toString() {
        return fileName + ":" + getLineNumber();
    }
    
    public boolean equals (Object obj) {
        SourceLocation other = (SourceLocation)obj;
        if (other == null)
            return false;
        else if (getLineNumber() != other.getLineNumber())
            return false;
        else if (fileName == null || other.fileName == null)
            return false;
        else return fileName.equals(other.fileName);
    }


    /**
     * @return the fileName
     */
    public String getFileName() {
        return fileName;
    }


    /**
     * @param fileName the fileName to set
     */
    public void setFileName(String fileName) {
        this.fileName = fileName;
    }


    /**
     * @return the lineNumber
     */
    public int getLineNumber() {
        return lineNumber;
    }


    /**
     * @param lineNumber the lineNumber to set
     */
    public void setLineNumber(int lineNumber) {
        this.lineNumber = lineNumber;
    }


}
